portal-api/src/crypto/utils.js

```diff
@@ -19,7 +19,7 @@ const PRIV_KEY = Buffer.from(process.env.JWT_SIGNING_KEY, 'base64').toString('as
 function validPassword(password, hash, salt) {
   const hashVerify = crypto.pbkdf2Sync(password, salt, 10000, 64, 'sha512').toString('hex');
 
-  return hash === hashVerify;
+  return crypto.timingSafeEqual(hash, hashVerify);
 }
```

Comment:
I like that we're thinking to change this but it also makes me a bit nervous because `timingSafeEqual` throws if `hash` and `hashVerify` are not the same length, and user passwords have a weird edge case where new users don't have a password set to start with - is that going to cause us to throw here, or does the code not get this far if the user doesn't have a hash in the db yet?

Reply:
Also, shall we change it in tfm-api too?

Reply:
It is indeed going to throw here -- I think a solution is to try catch here, and return false in the catch. This would ignore the time safe equals (https://github.com/nodejs/node/issues/17178) but I feel this is not an issue

Reply:
(this implimentation is not worse than the previous implimentation, dispite the try/catch block here)